COMETwebapp/COMETwebapp/Components/OpenModel.razor

```diff
@@ -19,98 +19,110 @@ Copyright (c) 2022 RHEA System S.A.
+            <div class="col-md-15 m-top-3px">
+                <InputSelect id="iteration" class="form-control" @bind-Value="SelectedModelDto.SelectedIterationSetupNumber">
+                    <option value="">--Select--</option>
+                    @foreach (var iteration in this.SelectedEngineeringModelSetup.IterationSetup)
                     {
-                        <option value="@iteration.IterationNumber">iteration_@iteration.IterationNumber - Active</option>
-                    } else
+                         @if(@iteration.FrozenOn != null)
+                        {
+                            <option value="@iteration.IterationNumber">iteration_@iteration.IterationNumber - @iteration.FrozenOn</option>
+                        } else
+                        {
+                            <option value="@iteration.IterationNumber">iteration_@iteration.IterationNumber - Active</option>
+                        }  
+                    }
+                </InputSelect>
+            </div>
+        </div>
+    }
+    @if(this.Domains.Count != 0)
+    {
+        <div class="form-group row">
+            <label for="domain">Select a Domain:</label>
+            <div class="col-md-15 m-top-3px">
+                <InputSelect id="domnain" class="form-control" @bind-Value="SelectedModelDto.SelectedDomainOfExpertiseName">
+                    <option value="">--Select--</option>
+                    @foreach (var domain in Domains)
                     {
-                        <option value="@iteration.IterationNumber">iteration_@iteration.IterationNumber - @iteration.FrozenOn</option>
+                        <option value="@domain.Name">@domain.Name</option>
                     }
-            }
-        </select>
-    </div>
-</div>
-}
-@if (Domains.Count != 0)
-{
-   <div class="form-group row">
-    <label>Domain:</label>
-    <div class="col-md-15 m-top-3px">
-        <select @onchange="SetDomain">
-            <option value="">--Select--</option>
-            @foreach(var domain in Domains)
-            {
-                <option value="@domain.Name">@domain.Name</option>
-            }
-        </select>
-    </div>
-</div>
-}
+                </InputSelect>
+            </div>
+        </div>
+    }
+    @if (this.SelectedEngineeringModelSetup != null && this.Domains.Count != 0)
+    {
+        <div class="row">
+            <div class="col-md-12 text-center">
+                <button type="submit" class="btn btn-connect">Open the Iteration</button>
+            </div>
+        </div> 
+    }
+</EditForm>
 
-@if (SelectedEngineeringModelSetup != null && SelectedIterationSetup != null && ISessionAnchor.CurrentDomainOfExpertise != null)
-{
-    <div class="row">
-        <button class="btn btn-connect " @onclick="OpenAModel"> Open</button>
-    </div>
-}
+@code{
+    /// <summary>
+    /// Class to define the object to set in EditForm
+    /// </summary>
+    public class ModelDto
```

Comment:
I think it is good practice for every class to reside in it's own .cs file so it is reusable.
On the other hand, do you really need a separate class? Not just 3 variables?
You define a ModelDto (sounds pretty generic, maybe even too generic) that has 3 properties that all start with the text "Selected".
So to me it looks like a generic class, that has very context-specific properties. 

I think you should think about making the ModelDto class more specific to the situation, or move the properties from ModelDto to the @code level.

I also think that it is not a DTO, or am I missing something here?
We use the term DTO for objects used to transfer data to an from a database.